File: FlexServ Backend/SpringBoot/TransactionService/src/main/java/com/flexserv/exception/ApiError.java
package com.flexserv.exception;

import java.time.LocalDateTime;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ApiError {
    private Boolean success = false;
    private String message;
    private LocalDateTime timestamp = LocalDateTime.now();

    public ApiError(String message) {
        this.message = message;
    }
}

File: FlexServ Backend/DotNet/TransactionService/Exceptions/ApiError.cs
namespace TransactionService.Exceptions;

public class ApiError
{
    public bool Success { get; set; } = false;
    public string Message { get; set; } = string.Empty;
    public DateTime Timestamp { get; set; } = DateTime.Now;

    public ApiError() { }

    public ApiError(string message)
    {
        Message = message;
    }

    public ApiError(bool success, string message, DateTime timestamp)
    {
        Success = success;
        Message = message;
        Timestamp = timestamp;
    }
}
